pygeoapi/provider/postgresql.py

```diff
@@ -382,80 +191,206 @@ def get(self, identifier, **kwargs):
 
         :returns: GeoJSON FeaturesCollection
         """
+        LOGGER.debug(f'Get item by ID: {identifier}')
+
+        # Execute query within self-closing database Session context
+        with Session(self._engine) as session:
+            # Retrieve data from database as feature
+            query = session.query(self.table_model)
+            item = query.get(identifier)
+            if item is None:
+                msg = f"No such item: {self.id_field}={identifier}."
+                raise ProviderItemNotFoundError(msg)
+            feature = self._sqlalchemy_to_feature(item)
+
+            # Add fields for previous and next items
+            id_field = getattr(self.table_model, self.id_field)
+            prev_item = (session.query(self.table_model)
+                         .order_by(id_field.desc())
+                         .filter(id_field < identifier)
+                         .first())
+            next_item = (session.query(self.table_model)
+                         .order_by(id_field.asc())
+                         .filter(id_field > identifier)
+                         .first())
+            feature['prev'] = (getattr(prev_item, self.id_field)
+                               if prev_item is not None else identifier)
+            feature['next'] = (getattr(next_item, self.id_field)
+                               if next_item is not None else identifier)
+
+        return feature
+
+    def _store_db_parameters(self, parameters):
+        self.db_user = parameters.get('user')
+        self.db_host = parameters.get('host')
+        self.db_port = parameters.get('port', 5432)
+        self.db_name = parameters.get('dbname')
+        self.db_search_path = parameters.get('search_path', ['public'])
+        self._db_password = parameters.get('password')
+
+    def _create_engine(self):
```

Comment:
This function would significantly impact the flexibility of our database connections since`SQLAlchemy` supports different databases. Perhaps it is worth moving it again to an abstract base class `DatabaseConnection` in a different PR

Reply:
Yes.  In theory the only difference between database providers is the connection string, so you could have an abstract `SQLAlchemyProvider` and the only thing you would need to override in the children would be some kind of `self._get_connection_string(parameters)` function.

It's a shame that GeoAlchemy2 doesn't support Oracle spatial.  Now that the cx_Oracle driver has become [python-oracledb](https://oracle.github.io/python-oracledb/) and doesn't need an Instant Client installation, it would have been trivial to add Oracle support.  It is still possible, but you'd need to do some trickery to export the geometry as WKT.

Yes, let's save this for a different PR.  After next week we won't have time to work on this for a while so it would be good to get this PR merged first.